gateway-js/src/index.ts

```diff
@@ -467,6 +445,102 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private async updateCsdl(result: CsdlUpdate): Promise<void> {
+    if (result.id === this.compositionId) {
+      this.logger.debug('No change in composition since last check.');
+      return;
+    }
+
+    const previousSchema = this.schema;
+    const previousCsdl = this.parsedCsdl;
+    const previousCompositionId = this.compositionId;
+
+    if (previousSchema) {
+      this.logger.info('Updated CSDL was found.');
+    }
+
+    await this.maybePerformServiceHealthCheck(result);
+
+    this.compositionId = result.id;
+    this.parsedCsdl = parse(result.csdl);
+
+    if (this.queryPlanStore) this.queryPlanStore.flush();
+
+    const { schema, composedSdl } = this.createSchema({
+      csdl: result.csdl,
+    });
+
+    if (!composedSdl) {
+      this.logger.error(
+        "A valid schema couldn't be composed. Falling back to previous schema.",
+      );
+    } else {
+      this.schema = schema;
+      this.queryPlannerPointer = getQueryPlanner(composedSdl);
```

Comment:
I've heard rumors that you can only have one query planner per process or something. Can you tell me more about this? Like, does calling this function immediately break the previous query planner?

Reply:
I think I see where this is going. To my knowledge, yes this will break the previous query planner.

Does this mean we have a problem handling schema rollover / existing in-flight operations? I think it does. Let me know if you have any thoughts. I'll open a separate issue for this.

Reply:
Yeah, I suspect there is a bug in the current codebase here... we do have https://github.com/apollographql/federation/issues/210 at least.